Map sidesets for changing mesh using mask

vanwdani opened the pull request:
Refs #23557



## Reason
Add sidesets, but not exclusively around subdomains

## Design
Minimal change to code, with only an additional inputParameter.

## Impact
Enhance API for more robust usage.

vanwdani committed:
Map sidesets for changing mesh using mask
Refs #23557

Update framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md

Co-authored-by: Daniel Schwen <daniel@schwen.de>

dschwen commented on framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md:
```diff
@@ -2,7 +2,10 @@
 ## Description
 
-The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_su ...
+The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_su ...
```
```suggestion
The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_subdomains` or that have no neighbor. This dynamic update can be useful in conjunction with subdomain update user objects such as [CoupledVarThresholdElementSubdomainModifier](CoupledVarThresholdElementSubdomainModifier.md).  The element sides on which this object acts can be limited using the [!param](/MeshModifiers/SidesetAroundSubdomainUpdate/mask_side) parameter, as shown in:
```

dschwen commented on framework/src/meshmodifiers/SidesetAroundSubdomainUpdater.C:
```diff
@@ -150,6 +163,8 @@ SidesetAroundSubdomainUpdater::finalize()
     for (auto & [elem_id, side] : sent_data)
     {
       _boundary_info.add_side(mesh.elem_ptr(elem_id), side, _boundary_id);
+      _boundary_info.clear_boundary_node_ids();
+      _boundary_info.build_node_list_from_side_list();
```
Is this safe? What if the user creates a custom node set (e.g. a single node to suppress rigid body translation). That would just be deleted and not replaced, because not every node set can be represented by an equivalent side set.

vanwdani replied:
`MooseMesh` defines `construct_node_list_from_side_list = true` by default.  Is there already another mechanism within Moose that adds to the list for just node sets?

dschwen replied:
But that doesn't _delete_ existing node sets!

loganharbour replied:
As long as this doesn't _remove_ any sidesets/nodesets, the clear shouldn't be necessary

dschwen replied:
It will shrink sidesets.

loganharbour replied:
What is _it_? The build call or this modifier? Because the build call only adds

vanwdani replied:
It seems like creating the nodesets with only a single node is the one-off case since most nodesets are constructed from sidesets.  Wouldn't this call for a seperate fix for this?

roystgnr replied:
Most nodesets in MOOSE are constructed from sidesets, but not all, and removing any other nodesets is definitely a bug.

Most nodesets in other software are specifically discrete sets of boundary points unrelated to sets of boundary sides.  MOOSE is only an exception here because back in 2009 I still reviewed libMesh contributions with a mindset of "does this new API do what it says it does" rather than also wondering "what XY Problem do I need to address before it's too late".

You might consider removing node BCIDs on a side's nodes whenever you remove that BCID on that side; then you could rerun `build_node_list_from_side_list()` to restore any overzealous removals without having to clear anything first.

dschwen replied:
The issue that remains is that if users explicitly set `construct_node_list_from_side_list = false` in their mesh block, they will get an unexpected construction of those node sets later on by this UO. But Roy's suggestion is a good compromise between being correct and easy to implement....

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/2837036/), step Docs: sync website on 49cf86b wanted to post the following:

View the site [here](https://mooseframework.inl.gov/docs/PRs/30388/site)

This comment will be updated on new commits.

vanwdani committed:
correct documentation link

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/2843327/), step Generate coverage on 1590a4c wanted to post the following:


## Framework coverage

<table>
  <thead>
    <tr>
      <th></th>
      <th>1172f9</th>
      <th colspan=4>#30388 1590a4</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>85.41%</td>
      <td>85.44%</td>
      <td>+0.03%</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>112323</td>
      <td>112369</td>
      <td>+46</td>
      <td>11</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>19190</td>
      <td>19153</td>
      <td>-37</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/30388/coverage_diff/framework)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/30388/coverage/framework)


## Modules coverage

Coverage did not change

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/contact)
- [`electromagnetics`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/electromagnetics)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/geochemistry)
- [`heat_transfer`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/heat_transfer)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/navier_stokes)
- [`optimization`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/optimization)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/richards)
- [`scalar_transport`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/scalar_transport)
- [`solid_mechanics`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/solid_mechanics)
- [`solid_properties`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/solid_properties)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/stochastic_tools)
- [`subchannel`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/subchannel)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/30388/coverage/xfem)

</details>



This comment will be updated on new commits.

vanwdani committed:
Remove remove nodesets

dschwen commented:
In remove functor use

```
  auto remove_functor =
      [this, &mesh, &displaced_mesh](const processor_id_type, const SideList & sent_data)
  {
    for (const auto & [elem_id, side] : sent_data)
    {
      const auto elem = mesh.elem_ptr(elem_id);
      _boundary_info.remove_side(elem, side, _boundary_id);
      for (const auto local_node_id: std::vector<unsigned int> elem->nodes_on_side(const unsigned int side))
        _boundary_info.remove_node(elem->node_ptr(local_node_id),  _boundary_id);
      if (_displaced_boundary_info)
      {
        const auto displaced_elem = displaced_mesh->elem_ptr(elem_id);
        _displaced_boundary_info->remove_side(displaced_elem , side, _boundary_id);
        for (const auto local_node_id: std::vector<unsigned int> displaced_elem->nodes_on_side(const unsigned int side))
          _displaced_boundary_info->remove_node(displaced_elem->node_ptr(local_node_id),  _boundary_id);
      }
    }
  };
```

dschwen commented:
We should probably check of the aforementioned option was set to `false` by the user and issue a warning in that case.

vanwdani committed:
Add test for construct_node_list_from_side_list

GiudGiud commented on framework/include/meshmodifiers/SidesetAroundSubdomainUpdater.h:
```diff
@@ -60,7 +60,7 @@ class SidesetAroundSubdomainUpdater : public DomainUserObject
   BoundaryID _boundary_id;
   ///@}
-
+  const BoundaryID _mask_side;
```
docstring

GiudGiud commented on framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md:
```diff
@@ -2,7 +2,10 @@
-The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_su ...
+The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_su ...
+
+!listing test/tests/mesh/add_sideset_ids/simple.i
+block=MeshModifiers
```
```suggestion
!listing test/tests/mesh/add_sideset_ids/simple.i block=MeshModifiers
```

GiudGiud commented on framework/src/meshmodifiers/SidesetAroundSubdomainUpdater.C:
```diff
@@ -160,10 +173,21 @@ SidesetAroundSubdomainUpdater::finalize()
     {
-      _boundary_info.remove_side(mesh.elem_ptr(elem_id), side, _boundary_id);
+      const auto elem = mesh.elem_ptr(elem_id);
+      if (!_mesh.getConstructNodeListFromSideList())
+        mooseWarning("The user has selected 'construct_node_list_from_side_list' as false, but "
+                     "SidesetAroundSubdomainUpdate is building node lists from the side lists.");
```
this will be hit thousands of times. Use mooseDoOnce too

GiudGiud commented on framework/src/meshmodifiers/SidesetAroundSubdomainUpdater.C:
```diff
@@ -160,10 +173,21 @@ SidesetAroundSubdomainUpdater::finalize()
+      {
+        const auto displaced_elem = displaced_mesh->elem_ptr(elem_id);
+        _displaced_boundary_info->remove_side(displaced_elem, side, _boundary_id);
+        for (const auto local_node_id : displaced_elem->nodes_on_side(side))
```
add a note to the documentation (!alert note) that nodesets are also updated

GiudGiud commented on test/tests/mesh/add_sideset_ids/tests:
```diff
@@ -31,4 +31,20 @@
+  [map_sidesets]
+    type = CSVDiff
+    input = simple.i
+    csvdiff = simple_out.csv
```
this CSV should include surface areas not just volumes

GiudGiud replied:
and number of nodes in the nodesets

vanwdani replied:
> and number of nodes in the nodesets

I tried using NodalSum, but the way the sideset works, the sideset doesn't exist on the mesh until after the `SidesetAroundSubdomainUpdater` switches it with the mask.

vanwdani replied:
Resolved

GiudGiud commented on test/tests/mesh/add_sideset_ids/tests:
```diff
@@ -31,4 +31,20 @@
+    expect_err ="The user has selected 'construct_node_list_from_side_list' as false, but SidesetAroundSubdomainUpdate is building node lists from the side lists."
+    design = MooseMesh.md
+    requirement ='The system shall throw a warning if the user has selected not to construct node list from side list while the SidesetAroundSubdomainUpdate does so.'
+    allow_warnings = false
```
```suggestion
```
that's the default

GiudGiud commented on test/tests/mesh/add_sideset_ids/tests:
```diff
@@ -31,4 +31,20 @@
+    input = simple.i
+    csvdiff = simple_out.csv
+    design = MooseMesh.md
+    requirement = 'The system shall be able to map sidesets as elements are ghosted/activated.'
```
```suggestion
    requirement = 'The system shall be able to map sidesets as the subdomain of elements are changed.'
```

activation is kind of in the context of your equation, let s focus on the mesh

GiudGiud commented on test/tests/mesh/add_sideset_ids/tests:
```diff
@@ -31,4 +31,20 @@
     design = MooseMesh.md
     requirement = 'The system shall be able to remove sides from sidesets when tese sides elements are moved to new blocks.'
   []
+  [map_sidesets]
```
These tests fits better in the subdmoain modifier folder imo

GiudGiud reviewed (changes requested):
nice new features!

vanwdani committed:
Changes after review

dschwen commented:
Typo aside, this might be a tricky one. Looks like the sideset assignment at this stage of the restore is incorrect (i.e. not consistent with the block assignment)

```
meshmodifiers/sideset_around_subdomain_updater.map_sidesets: *** ERROR ***
meshmodifiers/sideset_around_subdomain_updater.map_sidesets: 'right_noes' of type 'NodalSum' depends on variable(s) 'u'. However, that variable does not appear to be defined on (all of) boundary 'mask_right'.
```

vanwdani committed:
remove test part for pending PR

vanwdani committed:
Refine MooseMesh to count nodes correctly

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2857189/), step Clang format on 05c7401 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/30388/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/30388/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 6c09b202ee6e7650c2b8fc51946a790cfe6a444e`

GiudGiud commented on framework/src/meshmodifiers/SidesetAroundSubdomainUpdater.C:
```diff
@@ -160,10 +173,22 @@ SidesetAroundSubdomainUpdater::finalize()
     {
-      _boundary_info.remove_side(mesh.elem_ptr(elem_id), side, _boundary_id);
+      const auto elem = mesh.elem_ptr(elem_id);
+      if (!_mesh.getConstructNodeListFromSideList())
+        mooseDoOnce(mooseWarning(
+            "The user has selected 'construct_node_list_from_side_list' as false, but "
+            "SidesetAroundSubdomainUpdate is building node lists from the side lists."));
```
move this next to where the offending call is

maybe it should be a boolean option instead of being forced into it

GiudGiud commented on framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md:
```diff
@@ -2,10 +2,13 @@
 
 !alert note
-Use the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/execution_order_group) parameter to ensure that this user object is run after any subdomain changing user object has fully traversed the mesh.
+Use the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/execution_order_group) parameter to ensure that this user object is run after any subdomain changing user object has fully traversed the mesh.  Nodesets are also updated as well as sidesets, so the use of `construct_node_list_from_side_list = false` will not run as expected.
```
```suggestion
Use the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/execution_order_group) parameter to ensure that this user object is run after any subdomain changing user object has fully traversed the mesh. 

!alert warning
Nodesets are also updated as well as sidesets, so the use of `construct_node_list_from_side_list = false` will not run as expected.
```

GiudGiud commented on framework/src/mesh/MooseMesh.C:
```diff
@@ -489,6 +489,7 @@ MooseMesh::prepare(const MeshBase * const mesh_to_clone)
     const auto & add_sideset_ids = getParam<std::vector<BoundaryID>>("add_sideset_ids");
     _mesh_boundary_ids.insert(add_sideset_ids.begin(), add_sideset_ids.end());
     _mesh_sideset_ids.insert(add_sideset_ids.begin(), add_sideset_ids.end());
+    _mesh_nodeset_ids.insert(add_sideset_ids.begin(), add_sideset_ids.end());
```
no the param is add_sideset_ids. 

let's rename the parameter to add_boundary_ids / add_boundary_names

GiudGiud commented on framework/src/meshmodifiers/SidesetAroundSubdomainUpdater.C:
```diff
@@ -25,12 +25,18 @@ SidesetAroundSubdomainUpdater::validParams()
                                         "does not exist it will be added to the system.");
+  params.addParam<BoundaryID>("update_sideset_id",
+                              Moose::INVALID_BOUNDARY_ID,
+                              "The name of the sideset to be updated. If the boundary "
```
```suggestion
                              "The ID of the sideset to be updated. If the boundary "
```

GiudGiud commented on framework/include/meshmodifiers/SidesetAroundSubdomainUpdater.h:
```diff
@@ -59,6 +59,7 @@ class SidesetAroundSubdomainUpdater : public DomainUserObject
   ///@{ Boundary / sideset to update
   BoundaryName _boundary_name;
   BoundaryID _boundary_id;
+  const BoundaryID _mask_side;
   ///@}
```
```suggestion
  ///@}

  /// If specified, only sides also on this boundary will be updated
  const BoundaryID _mask_side;

```

GiudGiud commented on framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md:
```diff
@@ -2,10 +2,13 @@
 ## Description
 
-The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_su ...
+The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_su ...
```
let s explain this mask more. 

The mask is restricting the sides which can be added to the moving side as the boundary moves.
OR
The mask is restricting the sides which can be removed from the moving side as the boundary moves.

GiudGiud replied:
I think it s the former

GiudGiud commented on framework/src/meshmodifiers/SidesetAroundSubdomainUpdater.C:
```diff
@@ -25,12 +25,18 @@ SidesetAroundSubdomainUpdater::validParams()
   params.addParam<bool>("assign_outer_surface_sides",
                         true,
                         "Assign sides of elements im `inner_subdomains` that have no neighbor.");
   params.addRequiredParam<BoundaryName>("update_sideset_name",
                                         "The name of the sideset to be updated. If the boundary "
                                         "does not exist it will be added to the system.");
+  params.addParam<BoundaryID>("update_sideset_id",
```
either rename these two to boundary_id/name

OR 
add the corresponding ndoeset parameters

GiudGiud commented on test/tests/meshmodifiers/sideset_around_subdomain_updater/tests:
```diff
@@ -15,4 +15,19 @@
+    type = CSVDiff
+    input = simple.i
+    csvdiff = simple_out.csv
+    design = MooseMesh.md
+    requirement = 'The system shall be able to map sidesets as the subdomain of elements are changed.'
+  []
+  [sideset_warning]
+    type = RunException
+    input = simple.i
+    cli_args ='Mesh/construct_node_list_from_side_list=false'
+    expect_err ="The user has selected 'construct_node_list_from_side_list' as false, but SidesetAroundSubdomainUpdate is building node lists from the side lists."
+    design = MooseMesh.md
```
```suggestion
    design = SidesetAroundSubdomainUpdater.md
    requirement = 'The system shall be able to map sidesets as the subdomain of elements are changed.'
  []
  [sideset_warning]
    type = RunException
    input = simple.i
    cli_args ='Mesh/construct_node_list_from_side_list=false'
    expect_err ="The user has selected 'construct_node_list_from_side_list' as false, but SidesetAroundSubdomainUpdate is building node lists from the side lists."
    design = SidesetAroundSubdomainUpdater.md
```

GiudGiud commented on test/tests/meshmodifiers/sideset_around_subdomain_updater/tests:
```diff
@@ -15,4 +15,19 @@
+    cli_args ='Mesh/construct_node_list_from_side_list=false'
+    expect_err ="The user has selected 'construct_node_list_from_side_list' as false, but SidesetAroundSubdomainUpdate is building node lists from the side lists."
+    design = MooseMesh.md
+    requirement ='The system shall throw a warning if the user has selected not to construct node list from side list while the SidesetAroundSubdomainUpdate does so.'
```
```suggestion
    requirement ='The system shall throw a warning if the user has selected not to construct node list from side list but it does so anyway while updating sidesets.'
```

GiudGiud reviewed (changes requested):
I d like to avoid referring to nodesets as sidesets. 
We can use boundary for naming parameters, or we can make it more modular and have nodeset parameters. 
I like the latter better but the former is less effort (one line to rename a param)

vanwdani committed:
Update test/tests/meshmodifiers/sideset_around_subdomain_updater/tests

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

Update test/tests/meshmodifiers/sideset_around_subdomain_updater/tests

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

Update framework/include/meshmodifiers/SidesetAroundSubdomainUpdater.h

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

Update framework/src/meshmodifiers/SidesetAroundSubdomainUpdater.C

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

Update framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

Changes after review for naming

Update framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md

Co-authored-by: Guillaume Giudicelli <guillaume.giudicelli@gmail.com>

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2864392/), step Clang format on f1ad758 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/30388/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/30388/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 6c09b202ee6e7650c2b8fc51946a790cfe6a444e`

GiudGiud commented on framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md:
```diff
@@ -2,11 +2,17 @@
 Use the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/execution_order_group) parameter to ensure that this user object is run after any subdomain changing user object has fully traversed the mesh.
 
+!alert warning
+Nodesets are also updated as well as sidesets, so the use of `construct_node_list_from_side_list = false` will not run as expected.
```
```suggestion
Nodesets are also updated as well as sidesets, so the [!param](/Mesh/FileMesh/construct_node_list_from_side_list) `Mesh` parameter will not be respected.
```

GiudGiud commented on framework/src/mesh/MooseMesh.C:
```diff
@@ -486,19 +486,21 @@ MooseMesh::prepare(const MeshBase * const mesh_to_clone)
-    const auto & add_sideset_ids = getParam<std::vector<BoundaryID>>("add_sideset_ids");
-    _mesh_boundary_ids.insert(add_sideset_ids.begin(), add_sideset_ids.end());
-    _mesh_sideset_ids.insert(add_sideset_ids.begin(), add_sideset_ids.end());
+    const auto & add_boundary_ids = getParam<std::vector<BoundaryID>>("add_sideset_ids");
```
the parameter is still called sideset so there is still confusion between sidesets and ndoesets here

GiudGiud replied:
let's simply add the "add_nodeset_ids/names" capability

dschwen replied:
I think that's the right way of doing it. Side and node sets are fundamentally different, and we should offer separate parameters for them.

vanwdani replied:
I've added a lambda that handles the sidesets/nodesets.

GiudGiud commented on framework/src/meshmodifiers/SidesetAroundSubdomainUpdater.C:
```diff
@@ -25,12 +25,18 @@ SidesetAroundSubdomainUpdater::validParams()
                         true,
                         "Assign sides of elements im `inner_subdomains` that have no neighbor.");
-  params.addRequiredParam<BoundaryName>("update_sideset_name",
+  params.addRequiredParam<BoundaryName>("update_boundary_name",
```
rather than hard-deprecate, use the params.renameParam routine

vanwdani committed:
Refine Mesh for nodeset inputs

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2869987/), step Clang format on 747df55 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/30388/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/30388/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format fc26d03b7a90fb8d0acc819941d48eb1f3ea17cb`

vanwdani committed:
Add lambda for sideset/nodeset

Apply suggestions from code review

moosebuild commented:
Job [Precheck](https://civet.inl.gov/job/2890290/), step Clang format on 71fcfc5 wanted to post the following:

Your code requires style changes.  

A patch was auto generated and copied [here](https://mooseframework.inl.gov/docs/PRs/30388/clang_format/style.patch)  
You can directly apply the patch by running, in the top level of your repository:

`curl -s https://mooseframework.inl.gov/docs/PRs/30388/clang_format/style.patch | git apply -v`  

Alternatively, with your repository up to date and in the top level of your repository:

`git clang-format 603472b6a85e51ea67e11089c26a04e6696c3d07`

GiudGiud commented on framework/doc/content/source/meshmodifiers/SidesetAroundSubdomainUpdater.md:
```diff
@@ -2,11 +2,17 @@
 ## Description
 
-The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_su ...
+The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_su ...
```
```suggestion
The `SidesetAroundSubdomainUpdater` mesh modifier updates a sideset during a simulation by reassigning it based on a pair of subdomain sets, the [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/inner_subdomains) and [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/outer_subdomains). The sideset will be updated to comprise all sides along `inner_subdomains` that are neighboring `outer_subdomains` or that have no neighbor. This dynamic update can be useful in conjunction with subdomain update user objects such as [CoupledVarThresholdElementSubdomainModifier](CoupledVarThresholdElementSubdomainModifier.md).  

 An optional 'mask' can be used to restrict the sides which can be added to the moving side as the boundary moves, [!param](/MeshModifiers/SidesetAroundSubdomainUpdater/mask_side), as shown in:
```

GiudGiud commented on framework/src/mesh/MooseMesh.C:
```diff
@@ -186,6 +186,20 @@ MooseMesh::validParams()
+      "The listed nodeset names will be assumed valid for the mesh. This permits setting up "
+      "boundary restrictions for nodesets initially containing no sides. Ids for this nodesets may "
+      "be provided using add_nodesets_ids. In this case this list and add_nodesets_ids must "
+      "contain "
+      "the same number of items.");
```
```suggestion
      "contain the same number of items.");
```

GiudGiud commented on test/tests/meshmodifiers/sideset_around_subdomain_updater/tests:
```diff
@@ -15,4 +15,19 @@
+    input = simple.i
+    csvdiff = simple_out.csv
+    design = SidesetAroundSubdomainUpdater.md
+    requirement = 'The system shall be able to map sidesets as the subdomain of elements are changed.'
```
```suggestion
    requirement = 'The system shall be able to update sidesets defined by their location between two subdomains as the subdomain of elements are changed.'
```

GiudGiud reviewed (commented):
looks ready to me. I m going to add some distributed testing to make sure this passes once merged

GiudGiud commented:
Rebased and cleaned commit history. Planning to merge after tests
